client/app/components/packages/pas-form/edit.js

```diff
@@ -57,26 +58,32 @@ export default class PasFormComponent extends Component {
       isApplicantsDirty,
     } = this.pasForm;
 
-    return (isPasFormDirty && isPasFormValid) || isBblsDirty || isApplicantsDirty;
+    return !this.isUpdatingDb && ((isPasFormDirty && isPasFormValid) || isBblsDirty || isApplicantsDirty);
   }
 
   get isSubmittable() {
-    return this.submittableChanges.isValid;
+    return !this.isUpdatingDb && this.submittableChanges.isValid;
   }
 
-  @action
-  async save() {
-    await this.saveableChanges.save();
-
-    this.args.onSave();
+  get isUpdatingDb() {
+    return this.save.isRunning || this.submit.isRunning;
   }
 
-  @action
-  async submit() {
-    await this.submittableChanges.save();
+  @task(function* () {
+    yield this.saveableChanges.execute();
+    yield this.saveableChanges.rollback();
```

Comment:
i don't understand this part of the changesett API surface well enough so we'll wanna be vigilant about the side effects of this change...

Reply:
Agree. I haven't had extensive experience working with changeset rollbacks, either.